modules/imgcodecs/src/grfmt_jpeg2000_openjpeg.cpp

```diff
@@ -323,30 +323,28 @@ opj_dparameters setupDecoderParameters()
     return parameters;
 }
 
-opj_cparameters setupEncoderParameters(const std::vector<int>& params)
+opj_cparameters setupEncoderParameters(const std::vector<int>& params, const int numChannels)
 {
     opj_cparameters parameters;
     opj_set_default_encoder_parameters(&parameters);
-    bool rate_is_specified = false;
```

Comment:
It is possible to remove this temporary bool variable in original code if OpenCV default values are set before `params` are taken into the account.

Reply:
My intention was to avoid the default lossy compression but just found it a bit cleaner this way. At the end of the day, this is mainly a matter of style.